### ISSUE TITLE
DECISION PENDING - T4A Implementation

### ISSUE THREAD

**Issue by 2011sandeepraj** [bcgov/SIMS#119]
**Notes from February 7 meeting with Marisa**

**Implementation Timeline**
- Must be complete by Dec 31 of implementation year (PT vs FT irrelevant)
- Estimated lead time of 5-6 months for implementation
- Successful implementation is pre-requisuite for MNP sign-off
- Implementation requires testing component with NLSC / CAS/oracle (included within 6 month lead time estimate)

**Current Process**

**Requirements**
- Students need T4A which captures all income from Financial Aid received during a calendar year
- CAS/Oracle is reponsible for generating the T4A for the student; SABC is responsible for ensuring CAS/Oracle has the exhaustive list of funding received by the student within that calendar year. 
- CAS/Oracle is automatically aware of much of the student's funding without needing explicit action from SABC/BC. For e.g. _**[which funding types] are disbursed directly by NSLSC and are automatically sent to CAS by NSLSC**_
- CAS/Oracle is not aware of specific funding types, and needs SABC to go to extra steps to input those disbursements. This includes _**{which funding types]**_
- With an exhaustive list of disbursements to the student, CAS can produce a T4A and SABC can relay that to the student. 
- integration between SABC and CAS/Oracle occurs in two ways:
- Automatically for certain funding types including _**[which funding types]**_. For these funding types, CAS already has a supplier number for each student, and automatically associates those funding amounts with the student. 
- Manually by SBAC Finance staff inputting specific disbursements amounts into CAS
- CAS produces supplier numbers for students which represent the _Unique ID_ for each student. 

Every student reciving provincial disbursement through NSLSC needs supplier number. If no provincial grants and provincial funding, no CAS supplier number because 

Supplier maintenance portion and and update existing supplier numbers and push through to the accounts payale module accounting inputting into CAS. 

Anyone with provincial grant has a suplier number, SFAS sends disbursement report of all BCAG amounts being disbursed. Marisa rcords as an accountign expense. Next month, SFAS will pickup those disbursements for any students with Supplier number and records them in CAS so they are aware, but anyone without a supplier number needs to be manually entered in CAS by accounting department. 

Currently this is done on a montly interval through a batch (they're recorded in monthly batches). SIMS could do this differently - 

**How does the exchange/integration with CAS work?**
- SFAS talks with an intermediary system which then integates with CAS
- 

Under SIMS implementation, they will need to get the consolidated list of disbursements from BOTH SFAS and SIMS. 

Requirements:
- Associate a supplier number with each student requesting funding. I.e. we only need supplier numbers for each student receiving provincial disbursements and people with a PD/PPD flag. 
- CAS is parimarily concerned for being aware of disbursements for which they're not responsible for, so they can add them to their records and produce an accurate T4A. 
- CAS generates a supplier number for anyone new, or update the supplier number for anyone existing who has changed. 

**Additional Context**
- Some grants paid out to students are T4able (taxable) - Feds generate T4A slips
- SIMS will grant data and Feds will generate T4As for SIMS
-This is not needed for MVP
-FRCR can continue in parallel with T4A portion delivered later  (WHAT IS FRCR???)
-CAS team scheduled to deliver their portion of work by March 2023

**Acceptance Criteria**
- [ ] SIMS to connect to CAS Supplier # web service for real time information.
o Will work with CAS tech team to define what data is required from a SIMs perspective to get us Supplier # and…
- [ ] SIMS to connect to CAS Accounts Payable web service to send Account Payable file (Monthly/On demand?).
o Will work with CAS tech team to define what data is required from a SIMs perspective to process grant payment and generate T$As
- [ ] FRCR needs to be created

**Comment by HRAGANBC** [comment]
Closing - replaced by #3163 - #3169, which breaks apart the overall CAS integration problem (this ticket) into several smaller user stories and tickets
